plotly/tools.py

```diff
@@ -4559,6 +4563,23 @@ def _scatterplot_theme(dataframe, headers, diag, size, height,
                     return fig
 
     @staticmethod
+    def _hide_tick_labels_from_box_subplots(fig):
+        """
+        Hides tick labels for box plots in scatterplotmatrix subplots.
+        """
+        boxplot_xaxes = []
+        for j, plot_data in enumerate(fig['data']):
+            if plot_data['type'] == 'box':
+                boxplot_xaxes.append(
+                    'xaxis{}'.format(plot_data['xaxis'][-1])
+                )
+
+        for xaxis in boxplot_xaxes:
+            fig['layout'][xaxis]['showticklabels'] = False
+
+        return fig
```

Comment:
This makes a lot of sense. I `fig`ured that you'd need to assign back to `fig` cause was pointing to another fig and not mutating, but I suppose that's what it's doing.